File: contrib/jsimplemodule/example/Src.java
import org.omnetpp.simkernel.*;

public class Src extends JSimpleModule {
    private int i = 0;

/*
    public Src(long cptr) {
        super(cptr);
    }
*/

    protected void initialize() {
        ev().println("initialize of "+getFullPath());
        cMessage msg = new cMessage("timeout-"+getFullName());
        scheduleAt(simTime() + 1, msg);
    }

    protected void handleMessage(cMessage msg) {
        String msgname = "msg-" + i++;
        ev().println("sending "+msgname);
        send(new cMessage(msgname), "out");
        ev().println("scheduling next send");
        scheduleAt(simTime() + Simkernel.exponential(1), msg);
    }

    protected void finish() {
        ev().println("finish of "+getFullPath());
    }

};

File: test/anim/multisend/src.cc
#include <omnetpp.h>

using namespace omnetpp;

class Src : public cSimpleModule
{
  public:
    virtual void initialize() override;
    virtual void handleMessage(cMessage *msg) override;
    virtual ~Src() { }
};

Define_Module(Src);

void Src::initialize()
{
    scheduleAt(exponential(1.0), new cMessage("send"));
}

void Src::handleMessage(cMessage *msg)
{
    send(new cMessage("msg1"), "out1");
    send(new cMessage("msg2"), "out2");
    send(new cMessage("msg3"), "out1");

    scheduleAt(simTime()+exponential(1.0) + 0.01, msg);
}

